modules/solid_mechanics/include/materials/NewtonIterationSingleVariable.h

```diff
@@ -0,0 +1,214 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef NEWTONITERATIONSINGLEVARIABLE_H
+#define NEWTONITERATIONSINGLEVARIABLE_H
+
+#include "InputParameters.h"
+#include "ConsoleStream.h"
+
+/**
+ * Base class that provides capability for Newton return mapping
+ * iterations on a single variable
+ */
+
+class NewtonIterationSingleVariable;
```

Comment:
I had the same thought.  I had the idea originally that this would be something used for general single-variable Newton solves, and maybe it's possible to re-working it to be more general in the future.  I propose that for now, we keep this as a return mapping solver (which it appears can already be shared at least 4 different places), and then maybe revisit the idea of making it more general in the future.  I renamed it to SingleVariableReturnMappingSolution to reflect the fact that it is a return mapping solver. Since this is a base class, changing the name won't affect any input files.